### PR TITLE
Add Content-Type and Accept headers where necessary

### PR DESCRIPTION
**Pull request by provokateurin**
Closes https://github.com/stoplightio/elements/issues/1798

### CODE REVIEW

**Comment by mnaumanali94** [comment]
I think this should also close #2351

File: packages/elements-core/src/components/TryIt/build-request.ts [hunk 1]
```diff
@@ -300,3 +310,17 @@ function uriExpand(uri: string, data: Dictionary<string, string>) {
...
+function getAcceptedMimeTypes(httpOperation: IHttpOperation): string[] {
+  return [
+    ...new Set(
+      httpOperation.responses
+        .map(response =>
+          response === undefined || response.contents === undefined
+            ? []
+            : response.contents.map(contentType => contentType.mediaType),
+        )
+        .reduce((a, b) => [...a, ...b]),
```
Comment (dotslashderek) [inline packages/elements-core/src/components/TryIt/build-request.ts:323]: Currently, tests are failing on this line:

/home/circleci/project/packages/elements-core/src/components/TryIt/build-request.ts:237
            .reduce((a, b) => [...a, ...b])),
             ^

TypeError: Reduce of empty array with no initial value
    at Array.reduce (<anonymous>
    
    
I suspect you just need to change it to `.reduce((a, b) => [...a, ...b], [])` to give reduce an initial value to return in the case there's nothing to reduce

**Review by dotslashderek — COMMENTED** [review]
Very nice and appreciated - yes @mnaumanali94 I do think this will fix the other issue as well - @provokateurin I think the test failures are pretty straightforward to fix, take a look at my comment and thanks for contributing!

**Comment by provokateurin** [comment]
@dotslashderek Tests don't seem to run?

**Comment by dotslashderek** [comment]
@provokateurin yeah - we're trying to figure out what's going on with the CircleCI pipeline.  I pushed a couple changes to trigger runs, then fixed the four TryIt failures.  If you want to take a look at the HttpOperation test that's failing, I'd like to see what happens with the pipeline when you push a new commit.  Otherwise no worries I'll take another look at this in the next day or two.  Thanks again!

**Comment by provokateurin** [comment]
@dotslashderek I dropped your changes because I made a mistake with the Accept header where it still added it with 0 accepted mime types. Afterwards your fixes for the tests were breaking it again, so I did it properly. Tests work locally now.

File: packages/elements-core/src/components/TryIt/build-request.ts [hunk 2]
```diff
@@ -300,3 +314,17 @@ function uriExpand(uri: string, data: Dictionary<string, string>) {
...
+      httpOperation.responses
+        .map(response =>
+          response === undefined || response.contents === undefined
+            ? []
+            : response.contents.map(contentType => contentType.mediaType),
+        )
+        .reduce((a, b) => [...a, ...b], []),
+    ),
+  ];
+}
```
Comment (dotslashderek) [inline packages/elements-core/src/components/TryIt/build-request.ts:330]:
```suggestion

export function getAcceptedMimeTypes(httpOperation: IHttpOperation): string[] {
  return Array.from(
    new Set(
      httpOperation.responses.flatMap(response =>
        response === undefined || response.contents === undefined
          ? []
          : response.contents.map(contentType => contentType.mediaType),
      ),
    ),
  );
}
```

Comment (dotslashderek) [reply packages/elements-core/src/components/TryIt/build-request.ts:330]: I think the Array.from and flatMap are more readable - and flatMap will likely perform better than the map / reduce combo, although that might not matter given the limited number of responses in any given operation.

**Review by dotslashderek — COMMENTED** [review]
One comment, but a second request - could you add a couple quick unit tests around getAcceptedMimeTypes (that's why I `exported` it in the suggestion) to build-request.spec.ts? - could be as simple as:

```
describe('getAcceptedMimeTypes', () => {
    const operationSingleResponse: IHttpOperation = {
      id: 'adsf',
      method: 'GET',
      path: '/a/path',
      responses: [
        {
          id: 'responseA',
          code: '200',
          contents: [
            {
              id: 'adsf',
              mediaType: 'application/json',
            },
            {
              id: 'dfvs',
              mediaType: 'application/json',
            },
            {
              id: 'aaaa',
              mediaType: 'multipart/form-data',
            },
          ],
        },
      ],
    };

    const operationMultipleResponses: IHttpOperation = {
      id: 'sfdf',
      method: 'POST',
      path: '/a/nother/path',
      responses: [
        {
          id: 'responseA',
          code: '200',
          contents: [
            {
              id: 'adsf',
              mediaType: 'application/json',
            },
            {
              id: 'dfvs',
              mediaType: 'application/json',
            },
          ],
        },
        {
          id: 'responseB',
          code: '200',
          contents: [
            {
              id: 'adsf',
              mediaType: 'application/json',
            },
            {
              id: 'dfvs',
              mediaType: 'multipart/form-data',
            },
          ],
        },
        {
          id: 'responseC',
          code: '200',
          contents: [
            {
              id: 'dfvs',
              mediaType: 'multipart/form-data',
            },
          ],
        },
        {
          id: 'responseB',
          code: '200',
          contents: [
            {
              id: 'adsf',
              mediaType: 'text/json',
            },
            {
              id: 'dfvs',
              mediaType: 'multipart/form-data',
            },
          ],
        },
      ],
    };

    it('Handles a single response with duplicates', () => {
      expect(getAcceptedMimeTypes(operationSingleResponse)).toStrictEqual(['application/json', 'multipart/form-data']);
    });

    it('Handles multiple responses and dedups appropriately', () => {
      expect(getAcceptedMimeTypes(operationMultipleResponses)).toStrictEqual([
        'application/json',
        'multipart/form-data',
        'text/json',
      ]);
    });
  });
```

Thanks!  (And for fixing things properly w/ the TryIt tests 🙂 )

**Comment by provokateurin** [comment]
@dotslashderek Added the tests

**Review by dotslashderek — APPROVED** [review]
Looks good @provokateurin - thanks for taking care of this!

**Comment by provokateurin** [comment]
Missing reviews or ready to merge?

**Comment by provokateurin** [comment]
Thanks a lot!

**Pull request merged**